mandos-go/elrondgo-exporter/exportedTransaction.go

```diff
@@ -0,0 +1,148 @@
+package elrondgo_exporter
+
+import (
+	"bytes"
+	"encoding/hex"
+	"math/big"
+
+	"github.com/ElrondNetwork/arwen-wasm-vm/v1_4/arwen"
+	mj "github.com/ElrondNetwork/arwen-wasm-vm/v1_4/mandos-go/model"
+)
+
+const vmTypeHex = "0500"
+
+const dummyCodeMetadataHex = "0102"
+
+// length of "file:" in the mandos test
+const contractCodePrefixLength = 5
+
+type Transaction struct {
+	function   string
+	args       [][]byte
+	deployData []byte
+	nonce      uint64
+	value      *big.Int
+	esdtValue  []*mj.ESDTTxData
+	sndAddr    []byte
+	rcvAddr    []byte
+	gasPrice   uint64
+	gasLimit   uint64
+}
+
+func NewTransaction() *Transaction {
+	return &Transaction{
+		args:       make([][]byte, 0),
+		value:      big.NewInt(0),
+		esdtValue:  make([]*mj.ESDTTxData, 0),
+		sndAddr:    make([]byte, 0),
+		rcvAddr:    make([]byte, 0),
+		deployData: make([]byte, 0),
+	}
+}
+
+func (tx *Transaction) WithNonce(nonce uint64) *Transaction {
+	tx.nonce = nonce
+	return tx
+}
+
+func (tx *Transaction) GetNonce() uint64 {
+	return tx.nonce
+}
+
+func (tx *Transaction) WithCallValue(value *big.Int) *Transaction {
+	tx.value.Set(value)
+	return tx
+}
+
+func (tx *Transaction) GetCallValue() *big.Int {
+	return tx.value
+}
+
+func (tx *Transaction) WithESDTTransfers(esdtTransfers []*mj.ESDTTxData) *Transaction {
+	tx.esdtValue = append(tx.esdtValue, esdtTransfers...)
+	return tx
+}
+
+func (tx *Transaction) GetESDTTransfers() []*mj.ESDTTxData {
+	return tx.esdtValue
+}
+
+func (tx *Transaction) WithCallFunction(functionName string) *Transaction {
+	tx.function = functionName
+	return tx
+}
+
+func (tx *Transaction) GetCallFunction() string {
+	return tx.function
+}
+
+func (tx *Transaction) WithCallArguments(arguments [][]byte) *Transaction {
+	tx.args = append(tx.args, arguments...)
+	return tx
+}
+
+func (tx *Transaction) GetCallArguments() [][]byte {
+	return tx.args
+}
+
+func (tx *Transaction) WithSenderAddress(address []byte) *Transaction {
+	tx.sndAddr = make([]byte, 0)
+	tx.sndAddr = address
```

Comment:
You probably wanted this:
```suggestion
	tx.sndAddr = make([]byte, len(address))
	tx.sndAddr = copy(tx.sndAddr, address)
```